src/app/pages/LendingPage/components/LendingDialog/index.tsx

```diff
@@ -0,0 +1,267 @@
+
+          <FormGroup
+            label={t(translations.marginTradePage.tradeForm.labels.amount)}
+          >
+            <AmountInput
+              value={amount}
+              onChange={value => setAmount(value)}
+              asset={currency}
+              maxAmount={type === 'add' ? userBalance : depositedAssetBalance}
+            />
+          </FormGroup>
+
+          <div className="tw-mb-4 tw-mt-2">
+            {/* {type === 'add' && <AvailableBalance asset={currency} />} */}
+
+            {type === 'add' && (
+              <div
+                className={cn('tw-text-error tw-text-sm tw-text-center', {
+                  'tw-invisible tw-py-2 tw-mb-2': !errorMessage,
+                })}
+              >
+                {errorMessage}
+              </div>
+            )}
+
+            {type === 'remove' && (
+              <div className="tw-mb-10 tw-truncate tw-text-xs tw-font-light tw-tracking-normal">
+                <Trans
+                  i18nKey={
+                    translations.lendingPage.modal.withdraw.depositBalance
+                  }
+                  components={[
+                    <LoadableValue
+                      value={weiToNumberFormat(depositedAssetBalance, 6)}
+                      loading={loadingDepositedAssetBalance}
+                      tooltip={
+                        <>
+                          {fromWei(depositedAssetBalance)}{' '}
+                          <AssetRenderer asset={currency} />
+                        </>
+                      }
+                    />,
+                  ]}
+                />
+              </div>
+            )}
+          </div>
+
+          {type === 'add' && (
+            <>
+              <div
+                className={cn('tw-text-center tw-mt-8 tw-mb-12', {
+                  'tw-opacity-40': disabled(),
+                })}
+              >
+                <Trans
+                  i18nKey={translations.lendingPage.modal.deposit.apy}
+                  components={[
+                    <NextSupplyInterestRate
+                      asset={currency}
+                      weiAmount={lendingAmount}
+                    />,
+                  ]}
+                />
+              </div>
+            </>
+          )}
+
+          {/*<FormGroup label="Expected Reward:" className="tw-mb-5">*/}
+          {/*  <Input*/}
+          {/*    value="0"*/}
+          {/*    readOnly*/}
+          {/*    appendElem={<AssetRenderer asset={Asset.SOV} />}*/}
+          {/*  />*/}
+          {/*</FormGroup>*/}
+
+          <TxFeeCalculator
+            args={txFeeArgs}
+            methodName={methodName}
+            contractName={contractName}
+            className="tw-mt-6"
+          />
+
+          {/*{topupLocked?.maintenance_active && (*/}
+          {/*  <ErrorBadge content={topupLocked?.message} />*/}
+          {/*)}*/}
+
+          <DialogButton
+            confirmLabel={t(modalTranslation.cta)}
+            onConfirm={() => {
```

Comment:
This logic shouldn't be here. Create a method `handleSubmit` that will decide what handler it wants to call.